tests/integration_tests/performance/test_snapshot_restore_performance.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Performance benchmark for snapshot restore."""
+import json
+import logging
+import tempfile
+import pytest
+
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet, NetIfaceConfig
+from framework.builder import MicrovmBuilder, SnapshotBuilder, SnapshotType
+from framework.matrix import TestContext, TestMatrix
+from framework.stats import core
+from framework.stats.baseline import Provider as BaselineProvider
+from framework.stats.metadata import DictProvider as DictMetadataProvider
+from framework.utils import DictQuery
+from framework.utils_cpuid import get_cpu_model_name
+import host_tools.drive as drive_tools
+import host_tools.network as net_tools  # pylint: disable=import-error
+import framework.stats as st
+from integration_tests.performance.configs import defs
+from integration_tests.performance.utils import handle_failure, \
+    dump_test_result
+
+DEBUG = False
+TEST_ID = "snapshot_restore_performance"
+BASE_VCPU_COUNT = 1
+BASE_MEM_SIZE_MIB = 128
+BASE_NET_COUNT = 1
+BASE_BLOCK_COUNT = 1
+USEC_IN_MSEC = 1000
+
+# Measurements tags.
+RESTORE_LATENCY = "restore_latency"
+CONFIG = json.load(open(defs.CFG_LOCATION /
+                        "snap_restore_test_config.json"))
+
+# Define 4 net device configurations.
+net_ifaces = [NetIfaceConfig(),
+              NetIfaceConfig(host_ip="192.168.1.1",
+                             guest_ip="192.168.1.2",
+                             tap_name="tap1",
+                             dev_name="eth1"),
+              NetIfaceConfig(host_ip="192.168.2.1",
+                             guest_ip="192.168.2.2",
+                             tap_name="tap2",
+                             dev_name="eth2"),
+              NetIfaceConfig(host_ip="192.168.3.1",
+                             guest_ip="192.168.3.2",
+                             tap_name="tap3",
+                             dev_name="eth3")]
+
+# We are using this as a global variable in order to only
+# have to call the constructor and destructor once.
```

Comment:
Why shouldn't this be created alongside the test matrix, attached to its context and piped along to any scope that needs it?

Reply:
I had to revert this since no other option aside from the global variable seems to work. I tried putting it in the context and pass it along, and it would work for one kernel but on the second one the list would be unpopulated. I also tried generating it for every test that needed it without a global variable, but a local one, but then the destructor seems to get called before the test runs and the paths are invalid by the time the VM tries to use them. I must admit I don't actually know what exactly is causing this behavior, I am just guessing here. I did make the code that generates the mock drives prettier though :smile: